core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -118,6 +115,17 @@ func DbReceiptToEvmReceipt(receipt *dbReceipt) *evmtypes.Receipt {
 // Directly maps to onchain receipt schema.
 type rawOnchainReceipt = evmtypes.Receipt
 
+func (o *evmTxStore) Transaction(ctx context.Context, readOnly bool, fn func(*evmTxStore) error) (err error) {
+	var opts *sqlutil.TxOptions
+	if readOnly {
+		opts = &sqlutil.TxOptions{TxOptions: sql.TxOptions{ReadOnly: true}}
+	}
```

Comment:
I think this will be easier to read if we just set the field in either case:
```suggestion
	opts := &sqlutil.TxOptions{TxOptions: sql.TxOptions{ReadOnly: readOnly}}
```